crates/oxc_linter/src/rules/unicorn/no_array_reverse.rs

```diff
@@ -0,0 +1,164 @@
+use oxc_ast::{
+    AstKind,
+    ast::{ArrayExpressionElement, Expression},
+};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+use serde_json::Value;
+
+use crate::{AstNode, context::LintContext, rule::Rule};
+
+fn no_array_reverse_diagnostic(span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("Use `Array#toReversed()` instead of `Array#reverse()`.")
+        .with_help("`Array#reverse()` mutates the original array. Use `Array#toReversed()` to return a new reversed array without modifying the original.")
+        .with_label(span)
+}
+
+#[derive(Debug, Clone)]
+pub struct NoArrayReverse {
+    allow_expression_statement: bool,
+}
+
+impl Default for NoArrayReverse {
+    fn default() -> Self {
+        Self { allow_expression_statement: true }
+    }
+}
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Prefer using Array#toReversed() over Array#reverse().
+    ///
+    /// ### Why is this bad?
+    ///
+    /// Array#reverse() modifies the original array in place, which can lead to unintended side effects—especially
+    /// when the original array is used elsewhere in the code.
+    ///
+    /// ### Examples
+    ///
+    /// Examples of **incorrect** code for this rule:
+    /// ```js
+    /// const reversed = [...array].reverse();
```

Comment:
This is a bad example of `incorrect` code, because it will create a new array and does not modify the original one.

```suggestion
    /// const reversed = array.reverse();
```

I know that this is copied, but would like to have a better example ;) 
The same for the `correct` example

Reply:
ignore it for now: https://github.com/sindresorhus/eslint-plugin-unicorn/pull/2730